src/boot/shim.c

```diff
@@ -111,6 +126,10 @@ EFI_STATUS shim_load_image(
 void shim_retain_protocol(void) {
         uint8_t value = 1;
 
+        // TODO: drop setting this var once Shim < 16 is no longer supported, as the lock protocol is no longer needed
```

Comment:
hmm, wait a minute. this comment talks about the lock protocol, but the efi var below is not really related to the lock protocol?

or is it? if so, this function should probably exit immediately if shim_loaded() is false, and not install any efi var?

Reply:
It is related as the efi var ensures the old lock protocol is left behind after startimage(), that's its purpose. This is no longer necessary when the new protocol is available, as the old lock protocol won't be used.

Reply:
so why doesn't this function exit immediately if shim_loaded() is false? after all that seems to check for the lock proto, i.e. the thing this is about?

(i understand this is a pre-existing issue, but please fix it when you touch this)